### PR TITLE
45326644: Pull Request for Task 4 of COMP3710 Final Report (Topic Recognition)

### PR DESCRIPTION
**Pull request by navman1501**
Hello, 

My name is Navin. This is my lodging of the pull-request containing all the files and the commit log.
The task I completed was Task 4: "Segment the ISICs data set with the Improved UNet [1] with all labels having a minimum Dice similarity coefficient of 0.8 on the test set. [Normal Difficulty]". 

Please use the README file for more information.

### CODE REVIEW

**Comment by SiyuLiu0329** [comment]
- Sufficiently detailed commit messages
- Code valid for Normal Difficulty
- Good results, but the output the number of output classes being 2 is technically incorrect. The correct would have been 1 class + sigmoid or less correct 2 classes + softmax.

**Comment by rexsw** [comment]
Code looks good for normal difficulty

For next time;

Tensorflow has functions for shuffling and more so there’s no need to also use sklearn in your project. 

Comments such as 
    #Fit model onto training dataset with validation dataset as validation data
     results = unet.fit
Add nothing to the readability of the code and should be avoided. Overall comments were well done

File: recognition/ImprovedUNet_ISIC/main.py [hunk 1]
```diff
@@ -0,0 +1,167 @@
+#Import libraries
```
Comment (shakes76) [inline recognition/ImprovedUNet_ISIC/main.py:1]: Code header block should be added for all scripts/source files

**Comment by shakes76** [comment]
## TF Usage
OK
## Good Practice (Design/Commenting)
Good design and comments
## Algorithm 
Driver Script present
Shows Usage & Demo & Visualisation & Data usage, Readme OK
Module present
## Commit Log
Meaningful commit messages 
Progressive commits used 
## Documentation
Description and Comments good
Markdown used PDF submitted
## Pull Request
Successful Pull Request (Working Algorithm Delivered on Time in Correct Branch) 
**Feedback required. Please fix the conflict cause by the gitignore file, otherwise I can't merge.**
No Request Description. In the future, a more detailed pull request could've been done. When developer receiving the request is not known, the request could be rejected. Remember you could use/reuse the README info. 
Notebooks have markdown sections for you to explain what's going on, so you should make use of it.

File: recognition/ImprovedUNet_ISIC/main.py (hunk 1 above)
Comment (navman1501) [reply recognition/ImprovedUNet_ISIC/main.py:8]: Hey Shakes, I've added the code header block to all the scripts and source files.

**Comment by navman1501** [comment]
Hey Shakes & Siyu, I think I have addressed your feedback by:
- Adding the code header blocks
- Fixing the merge conflict
- Fixing the number of output classes to have 1 + sigmoid at the end. 

Thank you!
